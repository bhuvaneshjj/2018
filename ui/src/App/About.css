.About_row {
  display: flex;
  flex-direction: row;
}

.About_app {
  margin-left: 2em;
}

.About_timesync {
  display: flex;
  flex-grow: auto;
  margin-top: 2ex;
}
