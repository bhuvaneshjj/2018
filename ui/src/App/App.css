.App-div {
  height: 100vh;
  display: flex;
  flex-direction: column;
}

main {
  display: flex;
  flex: 1 0 auto;
  flex-direction: column;
}

/* Override react-bootstrap styling for Navbar's container. */
.navbar > .container {
  padding-left: 0px;
  margin-left: 0px;
}

/* Override react-bootstrap styling for Navbar's active item. */
.active > a {
  font-weight: bold;
}

/* Override react-boostrap styling for NavDropdown menu items. */
.dropdown-menu > li > a {
  padding-bottom: 6px;
  padding-top: 6px;
}

.App-Nav {
  font-size: 110%;
}

.App-logo,
.App-logo-animated {
  height: 50px;
}

.App-logo-animated {
  animation: App-logo-spin infinite 20s linear;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
