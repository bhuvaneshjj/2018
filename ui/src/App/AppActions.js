export const websocketConnected = () => ({ type: 'WEBSOCKET_CONNECTED' });

export const websocketDisconnected = () => ({ type: 'WEBSOCKET_DISCONNECTED' });
