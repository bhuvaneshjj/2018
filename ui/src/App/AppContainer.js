import { connect } from 'react-redux';
import AppDnD from './AppDnD';

const mapStateToProps = state => {
  const {
    auth,
    connected,
    timesync: { offset: timeOffset }
  } = state;

  return {
    authenticated: auth.authenticated,
    connected,
    timeOffset,
    username: (auth.decodedToken && auth.decodedToken.username) || null
  };
};

export default connect(
  mapStateToProps,
  null
)(AppDnD);
