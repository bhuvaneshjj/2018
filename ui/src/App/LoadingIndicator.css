.LoadingIndicator {
  animation: LoadingIndicator-spin infinite 5s linear;
}

@keyframes LoadingIndicator-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}