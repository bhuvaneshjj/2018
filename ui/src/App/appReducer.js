import { combineReducers } from 'redux';
import startujiciReducer from '../casomeric/Startujici/startujiciReducer';
import casomericReducer from '../casomeric/Casomeric/casomericReducer';
import registratorReducer from '../registrator/registratorReducer';
import rocnikyReducer from '../rocniky/rocnikyReducer';
import ucastniciReducer from '../ucastnici/ucastniciReducer';

const connected = (state = false, action) => {
  switch (action.type) {
    case 'WEBSOCKET_CONNECTED':
      return true;
    case 'WEBSOCKET_DISCONNECTED':
      return false;
    default:
      return state;
  }
};

const appReducer = combineReducers({
  casomeric: casomericReducer,
  startujici: startujiciReducer,
  registrator: registratorReducer,
  rocniky: rocnikyReducer,
  ucastnici: ucastniciReducer,
  connected
});

export default appReducer;
