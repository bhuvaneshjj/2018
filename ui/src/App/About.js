import React from 'react';
import PropTypes from 'prop-types';
import { Glyphicon, Panel, Well } from 'react-bootstrap';
import { version } from '../../package.json';
import logo from './logo.svg';
import './About.css';

const About = ({ username }) => (
  <Well>
    <Panel header="Jirkovský crossmarathon 2018" bsStyle="info">
      <div className="About_row">
        <img src={logo} className="App-logo-animated" alt="logo" />
        <div>
          Aplikace <Glyphicon glyph="star" />Jirkovský crossmarathon 2018<Glyphicon glyph="star" />
          <br />
          Verze: {version}
          <br />
          Origin: {(window && window.location && window.location.origin) || ''}
          <br />
          Přihlášen jako: {username}
        </div>
      </div>
    </Panel>
  </Well>
);

About.propTypes = {
  username: PropTypes.string
};

export default About;
