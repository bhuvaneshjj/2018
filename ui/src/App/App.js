import React from 'react';
import PropTypes from 'prop-types';
import { Redirect, Route, Switch } from 'react-router-dom';
import { Glyphicon, MenuItem, Nav, Navbar, NavDropdown, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import withAuth from '../auth/withAuth';
import withoutAuth from '../auth/withoutAuth';
import SignInContainer from '../auth/SignIn/SignInContainer';
import Casomeric from '../casomeric/Casomeric';
import withFetchUcastnici from '../entities/withFetchUcastnici/withFetchUcastnici';
import UcastniciDigestContainer from '../registrator/UcastniciDigest/UcastniciDigestContainer';
import Prihlasky from '../registrator/Prihlasky/Prihlasky';
import PrihlaseniContainer from '../registrator/Prihlaseni/PrihlaseniContainer';
import StartovniCislaContainer from '../registrator/StartovniCisla/StartovniCislaContainer';
import StartujiciContainer from '../registrator/Startujici/StartujiciContainer';
import UbytovaniContainer from '../registrator/Ubytovani/UbytovaniContainer';
import SignOutContainer from '../auth/SignOut/SignOutContainer';
import About from './About';
import { navsForMenu, navForRoute, navMenus } from './nav';
import './App.css';
import logo from './logo.svg';

const Item = ({ glyph, name }) => (
  <React.Fragment>
    <Glyphicon glyph={glyph} /> {name}
  </React.Fragment>
);
Item.propTypes = {
  glyph: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

const CurrentItem = ({ route }) => (
  <li className="App__current-item">
    <Item {...navForRoute({ route })} />
  </li>
);
CurrentItem.propTypes = {
  route: PropTypes.string.isRequired
};

const NavsForMenu = ({ menu, menuGlyph, menuKey }) => (
  <NavDropdown
    eventKey={menuKey}
    id={`app-dropdown-${menu}`}
    title={
      <span>
        <Glyphicon glyph={menuGlyph} /> {menu}
      </span>
    }
  >
    {navsForMenu({ menu }).map(({ key, glyph, name, path }) => (
      <LinkContainer key={key} to={path}>
        <MenuItem eventKey={key}>
          <Item glyph={glyph} name={name} />
        </MenuItem>
      </LinkContainer>
    ))}
  </NavDropdown>
);
NavsForMenu.propTypes = {
  menu: PropTypes.string.isRequired,
  menuGlyph: PropTypes.string.isRequired,
  menuKey: PropTypes.number.isRequired
};

const NavsAuthenticated = () =>
  navMenus.map(({ key, glyph, menu }) => (
    <NavsForMenu key={menu} menu={menu} menuGlyph={glyph} menuKey={key} />
  ));

const App = ({ authenticated, connected, location, username }) => (
  <div className="App__div">
    <Navbar inverse>
      <Navbar.Header>
        <img src={logo} className={connected ? 'App__logo--animated' : 'App__logo'} alt="logo" />
      </Navbar.Header>
      {authenticated && (
        <Nav className="App__Nav">
          <NavsAuthenticated />
          <CurrentItem route={location.pathname} />
        </Nav>
      )}
      <Nav className="App__Nav" pullRight>
        {!authenticated && (
          <LinkContainer to="/signin">
            <NavItem eventKey={5}>
              <Glyphicon glyph="log-in" /> Přihlášení
            </NavItem>
          </LinkContainer>
        )}
        {authenticated && (
          <LinkContainer key="signout" to="/signout">
            <NavItem eventKey={6}>
              <Glyphicon glyph="log-out" /> Odhlášení
            </NavItem>
          </LinkContainer>
        )}
        <LinkContainer to="/about">
          <NavItem eventKey={7}>
            <Glyphicon glyph="question-sign" /> O aplikaci
          </NavItem>
        </LinkContainer>
      </Nav>
    </Navbar>

    <main>
      <Switch>
        <Route exact path="/" component={withAuth(withFetchUcastnici(Casomeric))} />
        <Route path="/signin" component={withoutAuth(SignInContainer)} />
        <Route path="/casomeric" component={withAuth(withFetchUcastnici(Casomeric))} />
        <Route
          path="/ucastnici"
          component={withAuth(withFetchUcastnici(UcastniciDigestContainer))}
        />
        <Route path="/prihlasky" component={withAuth(withFetchUcastnici(Prihlasky))} />
        <Route path="/prihlaseni" component={withAuth(withFetchUcastnici(PrihlaseniContainer))} />
        <Route path="/startujici" component={withAuth(withFetchUcastnici(StartujiciContainer))} />
        <Route
          path="/startovni-cisla"
          component={withAuth(withFetchUcastnici(StartovniCislaContainer))}
        />
        <Route path="/ubytovani" component={withAuth(withFetchUcastnici(UbytovaniContainer))} />
        <Route path="/signout" component={withAuth(SignOutContainer)} />
        <Route path="/about" component={() => <About username={username} />} />
        <Redirect to="/" />
      </Switch>
    </main>
  </div>
);

App.propTypes = {
  authenticated: PropTypes.bool.isRequired,
  connected: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  }),
  username: PropTypes.string
};
App.defaultProps = {
  location: { pathname: '/' }
};

export default App;
