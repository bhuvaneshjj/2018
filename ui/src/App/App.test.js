import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from './App';

it('with websocket connected, authenticated', () => {
  const wrapper = shallow(<App authenticated={true} connected={true} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('with websocket disconnected, authenticated', () => {
  const wrapper = shallow(<App authenticated={true} connected={false} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('with websocket connected, not authenticated', () => {
  const wrapper = shallow(<App authenticated={false} connected={true} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
