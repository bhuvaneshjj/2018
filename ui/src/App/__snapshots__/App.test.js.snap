// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with websocket connected, authenticated 1`] = `
<div
  className="App__div"
>
  <Uncontrolled(Navbar)
    bsStyle="default"
    inverse={true}
  >
    <NavbarHeader>
      <img
        alt="logo"
        className="App__logo--animated"
        src="logo.svg"
      />
    </NavbarHeader>
    <Nav
      bsClass="nav"
      className="App__Nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <NavsAuthenticated />
      <CurrentItem
        route="/"
      />
    </Nav>
    <Nav
      bsClass="nav"
      className="App__Nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <LinkContainer
        activeClassName="active"
        exact={false}
        key="signout"
        replace={false}
        strict={false}
        to="/signout"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={6}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="log-out"
          />
           Odhlášení
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/about"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={7}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="question-sign"
          />
           O aplikaci
        </NavItem>
      </LinkContainer>
    </Nav>
  </Uncontrolled(Navbar)>
  <main>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/signin"
      />
      <Route
        component={[Function]}
        path="/casomeric"
      />
      <Route
        component={[Function]}
        path="/ucastnici"
      />
      <Route
        component={[Function]}
        path="/prihlasky"
      />
      <Route
        component={[Function]}
        path="/prihlaseni"
      />
      <Route
        component={[Function]}
        path="/startujici"
      />
      <Route
        component={[Function]}
        path="/startovni-cisla"
      />
      <Route
        component={[Function]}
        path="/ubytovani"
      />
      <Route
        component={[Function]}
        path="/signout"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Redirect
        push={false}
        to="/"
      />
    </Switch>
  </main>
</div>
`;

exports[`with websocket connected, not authenticated 1`] = `
<div
  className="App__div"
>
  <Uncontrolled(Navbar)
    bsStyle="default"
    inverse={true}
  >
    <NavbarHeader>
      <img
        alt="logo"
        className="App__logo--animated"
        src="logo.svg"
      />
    </NavbarHeader>
    <Nav
      bsClass="nav"
      className="App__Nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/signin"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={5}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="log-in"
          />
           Přihlášení
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/about"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={7}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="question-sign"
          />
           O aplikaci
        </NavItem>
      </LinkContainer>
    </Nav>
  </Uncontrolled(Navbar)>
  <main>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/signin"
      />
      <Route
        component={[Function]}
        path="/casomeric"
      />
      <Route
        component={[Function]}
        path="/ucastnici"
      />
      <Route
        component={[Function]}
        path="/prihlasky"
      />
      <Route
        component={[Function]}
        path="/prihlaseni"
      />
      <Route
        component={[Function]}
        path="/startujici"
      />
      <Route
        component={[Function]}
        path="/startovni-cisla"
      />
      <Route
        component={[Function]}
        path="/ubytovani"
      />
      <Route
        component={[Function]}
        path="/signout"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Redirect
        push={false}
        to="/"
      />
    </Switch>
  </main>
</div>
`;

exports[`with websocket disconnected, authenticated 1`] = `
<div
  className="App__div"
>
  <Uncontrolled(Navbar)
    bsStyle="default"
    inverse={true}
  >
    <NavbarHeader>
      <img
        alt="logo"
        className="App__logo"
        src="logo.svg"
      />
    </NavbarHeader>
    <Nav
      bsClass="nav"
      className="App__Nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <NavsAuthenticated />
      <CurrentItem
        route="/"
      />
    </Nav>
    <Nav
      bsClass="nav"
      className="App__Nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <LinkContainer
        activeClassName="active"
        exact={false}
        key="signout"
        replace={false}
        strict={false}
        to="/signout"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={6}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="log-out"
          />
           Odhlášení
        </NavItem>
      </LinkContainer>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/about"
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey={7}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="question-sign"
          />
           O aplikaci
        </NavItem>
      </LinkContainer>
    </Nav>
  </Uncontrolled(Navbar)>
  <main>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/signin"
      />
      <Route
        component={[Function]}
        path="/casomeric"
      />
      <Route
        component={[Function]}
        path="/ucastnici"
      />
      <Route
        component={[Function]}
        path="/prihlasky"
      />
      <Route
        component={[Function]}
        path="/prihlaseni"
      />
      <Route
        component={[Function]}
        path="/startujici"
      />
      <Route
        component={[Function]}
        path="/startovni-cisla"
      />
      <Route
        component={[Function]}
        path="/ubytovani"
      />
      <Route
        component={[Function]}
        path="/signout"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Redirect
        push={false}
        to="/"
      />
    </Switch>
  </main>
</div>
`;
